src/Exception/HostedDomainException.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace League\OAuth2\Client\Exception;
+
+/**
+ * Exception thrown if the Google Provider is configured with a hosted domain that the user doesn't belong to
+ */
+class HostedDomainException extends \Exception
+{
+    private $hostedDomainConfigured;
+
+    private $hostedDomainOfUser;
+
+    /**
+     * HostedDomainException constructor.
+     * @param string $hostedDomainConfigured
+     * @param string|null $hostedDomainOfUser
+     */
+    public function __construct($hostedDomainConfigured, $hostedDomainOfUser)
```

Comment:
Generally I like to see exceptions maintain the same constructor signature, with a static constructor:

```php
class HostedDomainException extends \Exception
{
    public static function notMatchingDomain($configuredDomain)
    {
        return new static("User is not part of domain $configuredDomain");
    }
}
```

This preserves the ability to use `$code` and `$previousException` when necessary.